src/main/daml/Daml/Finance/Interface/Instrument/Swap/Fpml/Types.daml

```diff
@@ -0,0 +1,214 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Interface.Instrument.Swap.Fpml.Types where
+
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (K)
+import Daml.Finance.Interface.Types.Common (InstrumentKey)
+import Daml.Finance.Interface.Types.Date.Calendar (BusinessDayConventionEnum)
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum)
+import Daml.Finance.Interface.Types.Date.RollConvention
+
+data Fpml = Fpml
```

Comment:
Could you help me understand what's the purpose of the `Fpml` data type? I see it is used in the `Factory` to create the `Instrument`, but it is not used in the instrument directly. 

The reason why I am asking is that there are daml-finance specific items (`InstrumentKey`, `lastEventTimestamp`) and what I assume is Fpml-specific terminology. I wonder if these should live in separate data types.

Reply:
It used to be the case that the interfaces had variables that matched the ones of the corresponding instrument. I believe this was changed in a refactoring recently. The interfaces now seem to contain one data structure (defined in the interface-specific Types file) containing the same variables. If you check out the other swaps / bonds you will see a similar pattern. Not sure what the motivation was for this design change, @brian-weir probably knows more if you are interested.

Reply:
In other words, I think the Fpml data type belongs where it is. Are you suggesting that the ones below (`SwapStream` etc) are moved to a different file? I thought about putting them in `Interface.Instrument.Types.Common`, but not sure yet whether they will be common to other workflows as well.

Reply:
I decided to move them to a new file `Interface.Instrument.Types.Fpml`. The file structure of the Fpml swap instrument now looks more like the others, and it also helps with package dependencies if you want reuse those types.

Reply:
Thank you, I understand the motivation now. 

The reason why I made a comment is the fact that when people read `Fpml`, they have a very specific data model in mind, which is the Fpml data model. But then we call a data type `Fpml` and add non-Fpml fields in there (e.g. `InstrumentKey`), which seems confusing. 

I agree with you that `SwapStream` etc ... will be used only for the `Fpml` swap, so I think the earlier location makes sense. However also moving them to a separate namespace is fine (e.g. the one you suggest) but I guess you should then create an additional package `Daml.Finance.Instrument.Types`.

My only concern in this comment was really that we are calling `Fpml` something which is not `Fpml`.

Reply:
Not sure we need another package for that. That has a build performance / maintainability impact as well. I think it is fine where it is for the time being. If you want we can discuss the package structure with the rest of the team in tomorrow's catchup and solve it as part of a separate PR if required.

Reply:
sounds good